compiler/haskell-ide-core/src/Development/IDE/Types/Diagnostics.hs

```diff
@@ -101,30 +101,20 @@ defDiagnostic _range _message = LSP.Diagnostic {
 
 -- | setLocation but with no range information
 dFilePath ::
-  Lens' LSP.Diagnostic (Maybe FilePath)
+  Lens' FileDiagnostic FilePath
 dFilePath = lens g s where
-    g :: LSP.Diagnostic -> Maybe FilePath
-    g d = (uriToFilePath . _uri) =<< view dLocation d
-    s :: LSP.Diagnostic -> Maybe FilePath -> LSP.Diagnostic
-    s d@Diagnostic{..} fp = set dLocation
-        (Location <$> (filePathToUri <$> fp) <*> pure _range) d
+    g = fst
+    s (_, d) fpB = (fpB, d)
 
 -- | This adds location information to the diagnostics but this is only used in
 --   the case of serious errors to give some context to what went wrong
 dLocation ::
```

Comment:
Yup, first pass I tried to change as little as possible and see what shook out. Turns out the lenses were one of those things. Should have used a draft PR to check the tests instead of a regular one